modules/core/src/test/java/org/apache/ignite/cache/IgniteCacheEntryProcessorSequentialCallTest.java

```diff
@@ -0,0 +1,108 @@
+import org.apache.ignite.transactions.TransactionIsolation;
+import javax.cache.processor.EntryProcessorException;
+import javax.cache.processor.MutableEntry;
+
+/**
+ */
+public class IgniteCacheEntryProcessorSequentialCallTest extends GridCommonAbstractTest {
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        return super.getConfiguration(igniteInstanceName)
+            .setCacheConfiguration(new CacheConfiguration("cache")
+                .setCacheMode(CacheMode.PARTITIONED)
+                .setAtomicityMode(CacheAtomicityMode.TRANSACTIONAL)
+                .setRebalanceMode(CacheRebalanceMode.SYNC)
+                .setWriteSynchronizationMode(CacheWriteSynchronizationMode.FULL_SYNC)
+                .setMaxConcurrentAsyncOperations(0)
+                .setBackups(0));
+    }
+
+    /**
+     * Test for sequential entry processor invoking not null value.
+     */
+    public void testSequentialCallOnOneNode() throws Exception {
+        Ignite ignite = startGrid(0);
+        try {
+            IgniteCache<TestKey, TestValue> cache = ignite.cache("cache");
+
+            TestKey key = new TestKey(1L);
+            TestValue val = new TestValue();
+            val.value("1");
+
+            cache.put(key, val);
+
+            NotNullCacheEntryProcessor cacheEntryProcessor = new NotNullCacheEntryProcessor();
+
+            try (Transaction transaction = ignite.transactions().txStart(TransactionConcurrency.OPTIMISTIC,
+                TransactionIsolation.SERIALIZABLE, 20000L, 1)) {
+
+                cache.invoke(key, cacheEntryProcessor);
+                cache.invoke(key, cacheEntryProcessor);
+
+                transaction.commit();
+            }
+        }
+        finally {
+            stopAllGrids();
+        }
+    }
+
+    /**
+     * Cache entry processor checking whether entry has got non-null value.
+     */
+    public static class NotNullCacheEntryProcessor implements CacheEntryProcessor<TestKey, TestValue, Object> {
+
+        /** {@inheritDoc} */
+        public Object process(MutableEntry entry, Object... arguments) throws EntryProcessorException {
+            assertNotNull(entry.getValue());
+
+            return null;
+        }
+    }
+
+    /**
+     *
+     */
+    public static class TestKey {
+        /** Value. */
+        private final Long val;
+
+        /**
+         * @param val Value.
+         */
+        public TestKey(Long val) {
+            this.val = val;
+        }
+    }
+
+    /**
+     *
+     */
+    public static class TestValue {
+        /** Value. */
+        private String val;
+
+        /**
+         * @return Value.
+         */
+        public String value() {
+            return val;
+        }
+
+        /**
+         * @param val New value.
+         */
+        public void value(String val) {
+            this.val = val;
+        }
+    }
+}
```

Comment:
Need empty line in the end of file.

Reply:
hmm....

Reply:
See point A. https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines#CodingGuidelines-A.ConfigureIntelliJIDEAcodestyle